bootstrap.py

```diff
@@ -918,6 +929,44 @@ class MEOptions:
         migrate_systems(options.org, options.activationkey)
         if options.enablerepos:
             enable_repos()
+    elif options.new_capsule and options.foreman_fqdn:
+        # > ELIF new_capsule and foreman_fqdn set, will migrate to other capsule
+        #
+        # > will replace CA certificate, reinstall katello-agent, gofer
+        # > update system definition to point to new capsule for content,
+        # > Puppet, OpenSCAP and update Puppet configuration
+        # > MANUAL SIGNING OF CSR OR MANUALLY CREATING AUTO-SIGN RULE STILL REQUIRED!
+        # > API doesn't have a public provision for creating auto-sign entries yet!
+        print_running("Removing katello-*, gofer and installing new Katello certificate")
+        clean_katello_agent()
+        exec_failexit("rpm -Uvh http://%s/pub/katello-ca-consumer-latest.noarch.rpm" % options.foreman_fqdn)
+        install_katello_agent()
+
+        print_running("Calling Foreman API to update content source, puppet master, puppet ca and openscap proxy records for %s" % FQDN)
+        capsule_id = return_matching_katello_key('capsules', 'name="%s"' % options.foreman_fqdn, 'id', False)
+        host_id = return_matching_foreman_key('hosts', 'name="%s"' % FQDN, 'id', False)
+
+        update_host_capsule_mapping(options.foreman_fqdn, API_PORT, "puppet_proxy_id", capsule_id, host_id)
+        update_host_capsule_mapping(options.foreman_fqdn, API_PORT, "puppet_ca_proxy_id", capsule_id, host_id)
+        update_host_capsule_mapping(options.foreman_fqdn, API_PORT, "content_source_id", capsule_id, host_id)
+        update_host_capsule_mapping(options.foreman_fqdn, API_PORT, "openscap_proxy_id", capsule_id, host_id)
+
+        print_running("Restarting goferd and rhsmcertd")
+        exec_failexit("/sbin/service rhsmcertd restart")
+        exec_failexit("/sbin/service goferd restart")
+
+        print_running("Stopping the Puppet agent for configuration update")
+        exec_failok("/sbin/service puppet stop")   # failok because people might be running Puppet from cron
+
+        print_running("Updating Puppet configuration")
+        exec_failexit("/usr/bin/puppet config set --section agent server %s" % options.foreman_fqdn)
```

Comment:
Yeah, I tried ConfigParser and it choked pretty badly on the default puppet.conf, so that is why we have this ugly blob to begin with.